### PR TITLE
Add common multi jvm config to cluster tests

### PR DESCRIPTION
**Pull request by chbatey**
Two of the cluster tests didn't have the common config which includes
not to shut down on leave

Refs #22632

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala [hunk 1]
```diff
@@ -186,7 +186,7 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
   /**
    * Scala API: Register a named entity type by defining the [[akka.actor.Props]] of the entity actor
    * and functions to extract entity and shard identifier from messages. The [[ShardRegion]] actor
-   * for this type can later be retrieved with the [[#shardRegion]] method.
+   * for this type can later be retrieved with the [[shardRegion]] method.
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: I know IntelliJ is complaining about these, but are they correct? Is that something new in ScalaDoc? It finds whatever it is? Does it work with genjavadoc?

File: akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingCustomShardAllocationSpec.scala [hunk 2]
```diff
@@ -153,7 +149,7 @@ abstract class ClusterShardingCustomShardAllocationSpec(config: ClusterShardingC
         runOn(first) {
           system.actorOf(Props[SharedLeveldbStore], "store")
         }
-        enterBarrier("peristence-started")
+        enterBarrier("persistence-started")
```
Comment (patriknw) [inline akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingCustomShardAllocationSpec.scala]: glad we got a native speaker in the team (or one with the spell checker turned on) ;-)

File: akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingLeavingSpec.scala [hunk 3]
```diff
@@ -81,7 +78,7 @@ abstract class ClusterShardingLeavingSpecConfig(val mode: String) extends MultiN
       dir = target/ClusterShardingLeavingSpec/sharding-ddata
       map-size = 10 MiB
     }
-    """))
+    """).withFallback(MultiNodeClusterSpec.clusterConfig))
```
Comment (patriknw) [inline akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingLeavingSpec.scala]: didn't I add the properties in `MultiNodeSpec`?
or is it `run-coordinated-shutdown-when-down = off` that is required here?

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala (hunk 1 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: I did google around for what scala doc supports and didn't find much: https://stackoverflow.com/questions/34491342/in-scaladoc-how-to-link-to-a-method-in-the-same-class

Without the # and step to in intellij works hence why i change some of them if i end up wanting to navigate there

**Review by ktoso — APPROVED** [review]

File: akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingLeavingSpec.scala (hunk 3 above)
Comment (chbatey) [reply akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingLeavingSpec.scala]: That was the property I was looking at but given these tests do an explicit leave the properties youa dded to `MultiNodeSpec` should be sufficient. Nearly all of our cluster tests do use that as a fallback but not the cluster sharding ones.

**Review by raboof — APPROVED** [review]

**Pull request merged**